server/src/actors/project.rs

```diff
@@ -429,7 +434,8 @@ impl Handler<GetEventAction> for Project {
         if now < self.retry_after {
             // Compensate for the missing subsec part by adding 1s
             let secs = (self.retry_after - now).as_secs() + 1;
-            return Response::ok(EventAction::RetryAfter(secs));
+            let reason = "TODO_ADD_REASON".to_string(); // TODO we need to also keep the reason
```

Comment:
Please keep the initial reason why this project was rate limited. Then, attach the reason instead of creating a new one.

For follow-up: We need to remember rate limits per project key separately. Please keep a map from public key to `(Instant, String)` for this.